src/blockdata/transaction.rs

```diff
@@ -618,6 +623,11 @@ impl SigHashType {
      pub fn as_u32(&self) -> u32 { *self as u32 }
 }
 
+impl From<SigHashType> for u32 {
```

Comment:
Hmm, how come you're allowed to do that here? I thought you could only make trait implementation "in the same crate as the definition of either the trait or the type implemented on"? I thought for this we needed `impl Into<u32> for SigHashType` instead. Interesting...

Reply:
Yeah this also surprises me.

Reply:
Orphan rules are more complex than that allowing things. My understanding is that since `SigHashType` is defined in the current crate, there can be no other crates that would be able to `From<SigHashType> for u32` as all types would be foreign to it. So it's allowed.

The best thing I could google about it RN is https://github.com/rust-lang/rfcs/issues/1856

Reply:
Hmm, this then somehow makes the assumption that "stdlib can never impl traits using types outside of stdlib", which is a special assumption that only holds for stdlib and not for other crates.

It does make sense though. I often implement `Into` in that case instead of `From`, but a `From` impl results in an `Into` impl but not the other way around IIRC. Even though `Into` is usually the one used in methods.